### PR TITLE
chore(ui): test that UI works behind proxy

### LINKED ISSUE #33705: [Bug]: UI Mode with subpath proxying does not report traces during test
### Version

1.49.0

### Steps to reproduce

Following on this issue: https://github.com/microsoft/playwright/issues/29564

Using the playwright UI to run any test.

In my case, the playwright UI is running inside a docker running VSCode's codeserver instance. When the UI mode is started (on port 9501), it is proxied on a path that looks like this: https://mydomain/proxy/9501/trace/uiMode.html

Hence all api requests made by playwright UI must include the prefix /proxy/9501, especially for the /trace endpoint. This is already the case for most requests made by UI mode, but not for trace requests happening during the execution of a test (the ones asking for a .json file that is generated by the trace server), where an absolute path is used instead, which prevents the UI from displaying ongoing progress in the test.

* `/trace/contexts?trace=...` use relative paths correctly --> `https://mydomain/proxy/9501/trace/contexts?trace=...`
* `/trace/file?path=my_trace.zip` use relative paths correctly. --> `https://mydomain/proxy/9501/trace/file?path=my_trace.zip`
* `/trace/file?path=my_trace.json` use an absolute path which makes them fail. --> `https://mydomain/trace/file?path=my_trace.json` (no `/proxy/9501`)

The issue only happens for network calls asking for a JSON trace (zip trace and contexts work)

I think the issue may come from here: https://github.com/microsoft/playwright/blob/fc19e6e7b4224bb9acecb89cbd73a8646a63eccf/packages/trace-viewer/src/sw/traceModelBackends.ts#L152
altough I'm not familiar enough with the code to be 100% sure.

In the sw.bundle.js file that is shipped with the playwright package, this is the line that causes the issue (there are 2 such lines):
```fetch("/trace/file?path="+encodeURIComponent(t))```

Replacing it by

```fetch("./file?path="+encodeURIComponent(t))```

In my case solves it, but I'm aware it's a fix inside a bundled file, which is not ideal.

### Expected behavior

Running a test in a playwright UI running behind a proxy should display test progress during the test.

### Actual behavior

Running a test in a playwright UI running behind a proxy does not display test progress during the test, but only reports on the status when the test ends. A bunch of network errors are seen in the network debug tab of chrome.

### Additional context

_No response_

### Environment

```shell
System:
    OS: Linux 5.10 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (4) x64 Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz
    Memory: 13.19 GB / 15.35 GB
    Container: Yes
  Binaries:
    Node: 20.15.1 - ~/workspace/.config/versions/node/v20.15.1/bin/node
    npm: 10.9.0 - ~/workspace/.config/versions/node/v20.15.1/bin/npm
  Languages:
    Bash: 5.1.16 - /usr/bin/bash
  npmPackages:
    @playwright/test: ^1.49.0 => 1.49.0
```

### PR DESCRIPTION
**Pull request by Skn0tt**
This PR adds a test to ensure we support running UI behind a proxy. It adds the `http-proxy` library to test this. As discussed, we need that for websocket support. I'll try replacing the existing `proxy` library with that in a separate PR.

Closes https://github.com/microsoft/playwright/issues/33705.

### CODE REVIEW

File: tests/playwright-test/ui-mode-trace.spec.ts [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 import { createImage } from './playwright-test-fixtures';
 import { test, expect, retries } from './ui-mode-fixtures';
+import http from 'node:http';
+import httpProxy from 'http-proxy';
+import { ManualPromise } from 'packages/playwright-core/lib/utils/manualPromise';
```
Comment (mxschmitt) [inline tests/playwright-test/ui-mode-trace.spec.ts:21]: We usually use relative imports I think, packages/playwright works by accident here I think.

**Review by mxschmitt — COMMENTED** [review]
Lets try to replace it first to see if its feasible?

File: tests/playwright-test/ui-mode-trace.spec.ts (hunk 1 above)
Comment (Skn0tt) [reply tests/playwright-test/ui-mode-trace.spec.ts]: this was a remnant, didn't even use it

**Comment by Skn0tt** [comment]
We discussed offline that we'd prefer not to have two proxy libraries within the project. But since `http-proxy` doesn't support `CONNECT`-style proxying and seems unmaintained, we can't use it as a replacement for `proxy`.

This PR is superceded by https://github.com/microsoft/playwright/pull/33771, which instead extends `proxy` with WebSocket support.

**Pull request closed without merging**